src/utils/Retriever.ts

```diff
@@ -0,0 +1,7 @@
+export type Retrievable<T> = T | (() => T);
+
+function isFunction<T>(x: any): x is () => T {
```

Comment:
(optional) you can add lodash or underscore if you want

Reply:
Currently the platform-client has `0` dependencies and I would like to keep it that way :slightly_smiling_face: